R/filter-nm-data.R

```diff
@@ -0,0 +1,193 @@
+#' Helper for reading in and parsing the `$DATA` record.
+#'
+#' @note This cannot be used for _modifying_ a `$DATA` record.
+#' @noRd
+#' @keywords internal
+read_data_record <- function(.mod){
+  data_recs <- get_records(.mod, "data")
+  n_data <- length(data_recs)
+  if(n_data !=1){
+    recs_fmt <- purrr::map_chr(data_recs, function(rec) rec$format())
+    rlang::abort(
+      c(
+        glue::glue("Expected a single data record, but found {n_data}:\n\n"),
+        recs_fmt
+      )
+    )
+  }
+  data_recs[[1]]$parse()
+  return(data_recs[[1]])
+}
+
+
+#' Extract IGNORE or ACCEPT options from a data record
+#' @param data_rec an `nmrec_record_data` object
+#' @inheritParams translate_nm_expr
+#' @noRd
+get_data_filter_exprs <- function(data_rec, type = c("ignore", "accept")){
+  type <- match.arg(type)
+
+  # Extract & format IGNORE/ACCEPT options
+  opts <- purrr::keep(data_rec$values, function(val){
+    inherits(val, "nmrec_option_value") && identical(val[["name"]], type)
+  })
+
+  exprs <- purrr::map_chr(opts, function(val){
```

Comment:
MIght be worth considering splitting by `,`  and unlisting here. Then you're guaranteed to have single expressions the rest of the way. I think you can avoid one `map()` call 
in `translate_nm_expr()`.